core/src/test/java/io/questdb/std/datetime/microtime/TimestampsTest.java

```diff
@@ -275,20 +245,54 @@ public void testFloorSS() throws Exception {
         TestUtils.assertEquals("2021-09-09T22:44:56.000Z", sink);
     }
 
-    @Test
-    public void testFloorYYYY() throws Exception {
-        long micros = TimestampFormatUtils.parseTimestamp("2008-05-12T23:45:51.045Z");
-        TimestampFormatUtils.appendDateTime(sink, Timestamps.floorYYYY(micros));
-        TestUtils.assertEquals("2008-01-01T00:00:00.000Z", sink);
-    }
-
     @Test
     public void testFloorWW() throws Exception {
         long micros = TimestampFormatUtils.parseTimestamp("2025-01-02T23:59:59.999Z");
         TimestampFormatUtils.appendDateTime(sink, Timestamps.floorWW(micros));
         TestUtils.assertEquals("2024-12-30T00:00:00.000Z", sink);
     }
 
```

Comment:
It would be good to add tests with pre-unix-epoch  timestamps . 
It seems that timestamp_floor behaves like timestamp_ceil for such : 
```sql
select timestamp_floor('7m', '1969-12-31T23:50:00.000000Z');
1969-12-31T23:53:00.000000Z
```
vs PG : 
```sql
SELECT date_bin('7 minute', '1969-12-31T23:50:00.000000', '1970-01-01 00:00:00.000000');
        date_bin
------------------------
 1969-12-31 23:46:00+01
```

Reply:
It is a tricky. We also don't check the pre-unix case for a "full unit":
```sql
select timestamp_floor('m', '1969-12-31T23:50:00.000000Z');
1969-12-31T23:50:00.000000Z
```
```sql
SELECT date_bin('1 minute', '1969-12-31T23:50:00.000000', '1970-01-01 00:00:00.000000');
1969-12-31 23:49:00+00
```
BTW, this was fixed in PG not too long ago.
So for now I would leave it as is or fix it all together.

Reply:
Yes, I noticed it's the same for basic units .
Could you create a GH issue for it ?